integrations/access/slack/status.go

```diff
@@ -0,0 +1,51 @@
+package slack
+
+import (
+	"net/http"
+
+	"github.com/gravitational/teleport/api/types"
+)
+
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+func statusFromStatusCode(httpCode int) types.PluginStatus {
+	code := types.PluginStatusCode_OTHER_ERROR
+	switch code {
+	case http.StatusUnauthorized:
+		code = types.PluginStatusCode_UNAUTHORIZED
+	default:
+		if httpCode >= 200 && httpCode < 400 {
+			code = types.PluginStatusCode_RUNNING
+		}
+	}
+	return &types.PluginStatusV1{Code: code}
+}
+
+func statusFromResponse(resp *APIResponse) types.PluginStatus {
+	if resp.Ok {
+		return &types.PluginStatusV1{Code: types.PluginStatusCode_RUNNING}
+	}
+
+	code := types.PluginStatusCode_OTHER_ERROR
+	switch resp.Error {
```

Comment:
Somewhat silly Qs: is Error guaranteed to be lower_camel like below? Could it have extraneous text?

Without knowing the Slack APIs this looks rather brittle, adding a comment would help.

Reply:
These are from the Slack API spec, see https://github.com/slackapi/slack-api-specs/blob/bc08db49625630e3585bf2f1322128ea04f2a7f3/web-api/slack_web_openapi_v2.json

As we do not use the Slack SDK, this is more or less the best we can do 🤷

Reply:
Thanks. Please leave a comment with the link in the code, it's very helpful.